### PR TITLE
e2e test enhancements for Stack Configuration Policy

### PR DESCRIPTION
**Pull request by kvalliyurnatt**
Adding e2e tests for the new Stack Config policy features that were added. Specifically the ability to add configs for Elasticsearch and KIbana and also configure secure settings for KIbana and additional secret mounts for Elasticsearch.

### CODE REVIEW

**Comment by barkbay** [comment]
buildkite test this -f p=gke,t=TestStackConfigPolicy,s=8.12.0-SNAPSHOT

File: test/e2e/es/stackconfigpolicy_test.go [hunk 1]
```diff
@@ -57,6 +58,8 @@ func TestStackConfigPolicy(t *testing.T) {
 
 	namespace := test.Ctx().ManagedNamespace(0)
 	secureSettingsSecretName := fmt.Sprintf("test-scp-secure-settings-%s", rand.String(4))
+	secretMountsSecretName := fmt.Sprintf("test-scp-secret-mounts-%s", rand.String(4))
```
Comment (barkbay) [inline test/e2e/es/stackconfigpolicy_test.go:61]: Let's maybe ensure that these 2 Secrets are eventually deleted?

**Comment by barkbay** [comment]
buildkite test this -f p=gke,t=TestStackConfigPolicyKibana,s=8.12.0-SNAPSHOT

File: test/e2e/kb/stackconfigpolicy_test.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+// TestStackConfigPolicy tests the StackConfigPolicy feature for Kibana.
+func TestStackConfigPolicyKibana(t *testing.T) {
+	// only execute this test if we have a test license to work with
+	if test.Ctx().TestLicense == "" {
+		t.SkipNow()
+	}
+
+	namespace := test.Ctx().ManagedNamespace(0)
+	// set up a 1-node Kibana deployment manually connected to Elasticsearch
```
Comment (barkbay) [inline test/e2e/kb/stackconfigpolicy_test.go:41]: Why "manually"?

File: test/e2e/kb/stackconfigpolicy_test.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+			},
+			test.Step{
+				Name: "Create a StackConfigPolicy",
+				Test: test.Eventually(func() error {
+					err := k.CreateOrUpdate(&policy)
+					return err
+				}),
+			},
+			test.CheckKeystoreEntries(k, KibanaKeystoreCmd, []string{"elasticsearch.pingTimeout"}, kbPodListOpts...),
+			kibanaChecks.CheckHeaderForKey(kbBuilder, "test", "kb-scp-test"),
```
Comment (barkbay) [inline test/e2e/kb/stackconfigpolicy_test.go:107]: nit: maybe add a comment to explain where does `"test"` come from.

File: test/e2e/kb/stackconfigpolicy_test.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+	esWithlicense := test.LicenseTestBuilder(esBuilder)
+
+	steps := func(k *test.K8sClient) test.StepList {
+		kibanaChecks := kibana.KbChecks{
+			Client: k,
+		}
+		return test.StepList{
+			test.Step{
+				Name: "Create a Secure Settings secret",
```
Comment (barkbay) [inline test/e2e/kb/stackconfigpolicy_test.go:93]: Same here, I think this is not deleted once it's no longer needed.

**Review by barkbay — APPROVED** [review]
LGTM

File: test/e2e/es/stackconfigpolicy_test.go (hunk 1 above)
Comment (kvalliyurnatt) [reply test/e2e/es/stackconfigpolicy_test.go:61]: I will add a step to delete these secrets during cleanup, these are mimicking user created secrets, so will not be deleted by the operator itself

File: test/e2e/kb/stackconfigpolicy_test.go (hunk 2 above)
Comment (kvalliyurnatt) [reply test/e2e/kb/stackconfigpolicy_test.go]: changed the comment.

File: test/e2e/kb/stackconfigpolicy_test.go (hunk 4 above)
Comment (kvalliyurnatt) [reply test/e2e/kb/stackconfigpolicy_test.go:93]: Added a step to delete the secrets

**Comment by kvalliyurnatt** [comment]
buildkite test this -f p=gke,t=TestStackConfigPolicy,s=8.12.0-SNAPSHOT

**Comment by kvalliyurnatt** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|2.11|[](https://github.com/elastic/cloud-on-k8s/pull/7452)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**